feat(gitlab): allow override server version (#11416)
renovatebot/renovate: docs/usage/self-hosted-experimental.md

If set to any value, Renovate will use a "hard" `process.exit()` once all work is done, even if a sub-process is otherwise delaying Node.js from exiting.
See <https://github.com/renovatebot/renovate/issues/8660> for background on why this was created.

> ## RENOVATE_X_SERVER_VERSION

Reviewer: ```suggestion
## RENOVATE_X_PLATFORM_VERSION
```

Wouldn't this naming be better?

> If set, renovate will use that string as GitLab server version instead of checking via GitLab api.

Reviewer 2: ```suggestion
If set, Renovate will use this string as GitLab server version instead of checking via the GitLab API.
```

> This can be usefull when using the GitLab `CI_JOB_TOKEN` for running renovate.

Reviewer 2: ```suggestion
This can be useful when using the GitLab `CI_JOB_TOKEN` for running Renovate.
```

> This can be usefull when using the GitlBal `CI_JOB_TOKEN` for running renovate.

Reviewer: ```suggestion
This can be usefull when using the GitLab `CI_JOB_TOKEN` for running renovate.
```

Reviewer: I think some expanded explanation here is needed. Have we documented elsewhere how/why we use the GitLab version string?

Author: We don't have any docs about version usage yet. Will add to `lib/platform/gitlab/index.md`

*Usages*
- PR `Draft:` prefix instead of `WIP:` (`>= v13.2.0`)
- Truncate markdown to 25K (`< v13.4.0`)
- Add reviewers (`>= v13.9.0`)

Reviewer: Another possibility would be an option called `skipPlatformCompatibility` that simply assumes everything is supported. i.e. not requiring someone to put a fixed version in their pipeline which could need changing regularly?

Author: It's not ment to be hard-coded, it's to use the `CI_SERVER_VERSION` variable in gitlab pipelines, with will be always defined 😉 

EG: set `RENOVATE_X_SERVER_VERSION: $CI_SERVER_VERSION` in pipeline.

> This can be useful when using the GitLab `CI_JOB_TOKEN` for running Renovate.

Reviewer 2: ```suggestion
This can be useful when you use the GitLab `CI_JOB_TOKEN` to authenticate Renovate.
```

Would this be better?

I don't think you're using the `CI_JOB_TOKEN` itself to _run_ Renovate, but to _authenticate_ to GitLab?

> Read [platform details](modules/platform/gitlab/index.md) to learn why we need the server version on GitLab.

Reviewer 2: I think there's some problem with the link to the `platform details`, I get a 404 page not found when I click the link in the GitHub preview.

Maybe we need a `../` in the path?

Author: No. the link should work when going live

> Checkout [lib/platform/gitlab/index.md](https://github.com/renovatebot/renovate/blob/HEAD/lib/platform/gitlab/index.md) for why we need the server version on GitLab.

Reviewer 2: ```suggestion
Read [lib/platform/gitlab/index.md](https://github.com/renovatebot/renovate/blob/HEAD/lib/platform/gitlab/index.md) to learn why we need the server version on GitLab.
```

> Read [lib/platform/gitlab/index.md](https://github.com/renovatebot/renovate/blob/HEAD/lib/platform/gitlab/index.md) to learn why we need the server version on GitLab.

Author: ```suggestion
Read [platform details](/modules/platform/gitlab/index.md) to learn why we need the server version on GitLab.
```

Link to new page, which will be available in a few minutes

Reviewer 2: I like the new platform details page, nice work! 😄 

You'll need to apply the suggestion yourself though. 😉
